src/Components/SingleArticle.jsx

```diff
@@ -24,6 +26,23 @@ export default function SingleArticle() {
     const clickShowComments = () => {
         setShowComments(true);
     }
+
+    const handleVote = (newVote) => {
+        patchVote(article_id, newVote)
+          .then(() => {
+            setSingleArticle((prevState) => ({
+              ...prevState,
+              votes: prevState.votes + newVote,
+            }));
+            setHasVoted(newVote);
+          })
+          .catch((error) => {
+            console.error(error);
+            window.alert('Vote update failed. Please try again');
```

Comment:
Great to see you looking at error handling here, but alerts aren't considered the best strategy to feedback statuses such as errors/success/loading states to the user for a number of reasons. From a UX standpoint they can be quite aggressive and cause irritation, but also from a usability standpoint a lot of users have pop-up blocker browser extensions nowadays which could prevent them from showing up at all. 

Instead I would have a look at the notes on error handling, and some of he ways we could use state to conditionally render error messages on our page. Think about how you might see errors shown to you on other websites for inspiration, what sort of handling do we commonly see happen if we enter our details incorrectly when making an online purchase on a shopping website for example. Bear in mind that there is a ticket for this later on so don't feel the need to refactor this now, you can come back to it later/try it on tickets going forwards! 😄 

Error handling notes: https://notes.northcoders.com/courses/js-front-end/error-handling